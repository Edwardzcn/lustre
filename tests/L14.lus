node nats() returns (n : int);
let
  n = 0 -> 1 + pre n;
tel

node f() returns (y : int);
  var c : bool;
  var x : int when c;
let
  c = true -> not pre c;
  x = nats () when c;
  y = current x;
tel

node main(puase : int) returns (y : int);
let
  --%MAIN;
  y = f();
tel


