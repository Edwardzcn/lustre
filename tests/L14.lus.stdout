node (puase : int)
  returns (y)
  assumes ()
  shows ()
let
  valof_pre___no_static : int
    = pre __no_static
   
  __no_static : int
    = 0 -> Add(1, valof_pre___no_static)
   
  valof_pre_c : bool
    = pre c
   
  c : bool
    = true -> Not(valof_pre_c)
   
  x : int when c
    = __no_static when c
   
  y : int
    = current x
tel
--- Step 1 ---
puase : int  = 0
y = 0
--- Step 2 ---
puase : int  = 0
y = 0
--- Step 3 ---
puase : int  = 0
y = 2
--- Step 4 ---
puase : int  = 0
y = 2
--- Step 5 ---
puase : int  = 0
y = 4
--- Step 6 ---
puase : int  = 0
y = 4
--- Step 7 ---
puase : int  = 0
y = 6
--- Step 8 ---
puase : int  = 0
y = 6
--- Step 9 ---
puase : int  = 0
y = 8
--- Step 10 ---
puase : int  = 0
y = 8
--- Step 11 ---
puase : int  = (EOF)
